crates/viewer/re_ui/src/notifications.rs

```diff
@@ -0,0 +1,393 @@
+use std::time::Duration;
+
+use egui::hex_color;
+use jiff::{Unit, Zoned};
+pub use re_log::Level;
+
+use crate::icons;
+use crate::UiExt;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+enum NotificationLevel {
+    Info,
+    Warning,
+    Error,
+    Success,
+}
+
+impl From<re_log::Level> for NotificationLevel {
+    fn from(value: re_log::Level) -> Self {
+        match value {
+            re_log::Level::Trace | re_log::Level::Debug | re_log::Level::Info => Self::Info,
+            re_log::Level::Warn => Self::Warning,
+            re_log::Level::Error => Self::Error,
+        }
+    }
+}
+
+fn is_relevant(level: re_log::Level) -> bool {
+    matches!(
+        level,
+        re_log::Level::Warn | re_log::Level::Error | re_log::Level::Info
+    )
+}
+
+pub fn notification_toggle_button(
+    ui: &mut egui::Ui,
+    show_notification_panel: &mut bool,
+    has_unread_notifications: bool,
+) {
+    let response = ui.medium_icon_toggle_button(&icons::NOTIFICATION, show_notification_panel);
+
+    if has_unread_notifications {
+        let mut pos = response.rect.right_top();
+        pos.x -= 2.0;
+        pos.y += 2.0;
+        ui.painter().circle_filled(pos, 3.0, hex_color!("#ab0037"));
+    }
+}
+
+struct Notification {
+    level: NotificationLevel,
+    text: String,
+
+    created_at: Zoned,
+    ttl: Duration,
+}
+
+pub struct NotificationUi {
+    /// State of every notification.
+    ///
+    /// Notifications are stored in order of ascending age, so the latest one is at the end.
+    data: Vec<Notification>,
```

Comment:
Also, `data` is one of those empty names (like "thing" or "stuff"). Better to call it `notifications` or `messages`